benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/TrivialMatcher.cs

```diff
@@ -10,16 +10,27 @@ namespace Microsoft.AspNetCore.Routing.Matchers
     // A test-only matcher implementation - used as a baseline for simpler
     // perf tests. The idea with this matcher is that we can cheat on the requirements
     // to establish a lower bound for perf comparisons.
-    internal class TrivialMatcher : Matcher
+    internal class TrivialMatcher : MatcherBase
     {
         private readonly MatcherEndpoint _endpoint;
+        private readonly Candidate[] _candidates;
+        private readonly int[] _candidateIndices;
+        private readonly int[] _candidateGroups;
 
         public TrivialMatcher(MatcherEndpoint endpoint)
         {
             _endpoint = endpoint;
+
+            _candidates = new Candidate[]
+            {
+                new Candidate(endpoint, Array.Empty<MatchProcessor>()),
+            };
+
+            _candidateIndices = new int[] { 0, };
+            _candidateGroups = new int[] { 1, };
         }
 
-        public override Task MatchAsync(HttpContext httpContext, IEndpointFeature feature)
+        public new Task MatchAsync(HttpContext httpContext, IEndpointFeature feature)
```

Comment:
Why is `new` needed here?

Reply:
The base class implementation is `sealed` by design. Adding a comment

Reply:
Actually I'll just remove the inheritance. There's no need for `MatcherBase` to exist once the instruction matcher is gone.